oio/crawler/integrity.py

```diff
@@ -122,11 +123,15 @@ def write_error(self, target, irreparable=False):
         self.error_writer.writerow(error)
 
     def write_rebuilder_input(self, target, obj_meta, irreparable=False):
-        ct_meta = self.list_cache[(target.account, target.container)][1]
+        cid = None
         try:
-            cid = ct_meta['system']['sys.name'].split('.', 1)[0]
+            ct_meta = self.list_cache[(target.account, target.container)][1]
+            try:
+                cid = ct_meta['system']['sys.name'].split('.', 1)[0]
+            except KeyError:
+                cid = ct_meta['properties']['sys.name'].split('.', 1)[0]
         except KeyError:
-            cid = ct_meta['properties']['sys.name'].split('.', 1)[0]
+            cid = cid_from_name(target.account, target.container)
```

Comment:
I "fixed" this with another solution in #1830. See https://github.com/open-io/oio-sds/pull/1830/files#diff-f925e1e5526264e38481ecb25a4486dbR58